thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/wrapper/GrouperWrapper.java

```diff
@@ -0,0 +1,90 @@
+
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.collections.MapUtils;
+import org.apache.pinot.thirdeye.datalayer.dto.DetectionConfigDTO;
+import org.apache.pinot.thirdeye.datalayer.dto.MergedAnomalyResultDTO;
+import org.apache.pinot.thirdeye.detection.ConfigUtils;
+import org.apache.pinot.thirdeye.detection.DataProvider;
+import org.apache.pinot.thirdeye.detection.DetectionPipeline;
+import org.apache.pinot.thirdeye.detection.DetectionPipelineResult;
+import org.apache.pinot.thirdeye.detection.DetectionUtils;
+import org.apache.pinot.thirdeye.detection.spi.components.Grouper;
+
+
+/**
+ * A group wrapper which triggers the configured grouper. The actual grouper
+ * must implement the {@link Grouper} interface.
+ */
+public class GrouperWrapper extends DetectionPipeline {
+  private static final String PROP_NESTED = "nested";
+  private static final String PROP_CLASS_NAME = "className";
+  private static final String PROP_GROUPER = "grouper";
+
+  private final List<Map<String, Object>> nestedProperties;
+
+  private final Grouper grouper;
+  private final String grouperName;
+
+  public GrouperWrapper(DataProvider provider, DetectionConfigDTO config, long startTime, long endTime)
+      throws Exception {
+    super(provider, config, startTime, endTime);
+    Map<String, Object> properties = config.getProperties();
+    this.nestedProperties = ConfigUtils.getList(properties.get(PROP_NESTED));
+
+    // Get the configured grouper from the components
+    Preconditions.checkArgument(this.config.getProperties().containsKey(PROP_GROUPER));
+    this.grouperName = DetectionUtils.getComponentName(MapUtils.getString(config.getProperties(), PROP_GROUPER));
+    Preconditions.checkArgument(this.config.getComponents().containsKey(this.grouperName));
+    this.grouper = (Grouper) this.config.getComponents().get(this.grouperName);
+  }
+
+  /**
+   * Runs the nested pipelines and calls the group method to group the anomalies accordingly.
+   * @return the detection pipeline result
+   */
+  @Override
+  public final DetectionPipelineResult run() throws Exception {
+    List<MergedAnomalyResultDTO> candidates = new ArrayList<>();
+    for (Map<String, Object> properties : this.nestedProperties) {
+      DetectionConfigDTO nestedConfig = new DetectionConfigDTO();
+
+      Preconditions.checkArgument(properties.containsKey(PROP_CLASS_NAME), "Nested missing " + PROP_CLASS_NAME);
+      nestedConfig.setId(this.config.getId());
+      nestedConfig.setName(this.config.getName());
+      nestedConfig.setDescription(this.config.getDescription());
+      nestedConfig.setProperties(properties);
+
+      DetectionPipeline pipeline = this.provider.loadPipeline(nestedConfig, this.startTime, this.endTime);
+
+      DetectionPipelineResult intermediate = pipeline.run();
+      candidates.addAll(intermediate.getAnomalies());
```

Comment:
Could you take the lastTimeStamp in the intermediate result and pass it to the next level? Otherwise, the detection water mark won't be updated. 

Also, pass the predictionResult for showing the predicted baselines and boundaries.

For example
```
Set<Long> lastTimeStamps = new HashSet<>();
List<PredictionResult> predictionResults = new ArrayList<>();
for{
// run nested  pipelines
DetectionPipelineResult intermediate = pipeline.run();
lastTimeStamps.add(intermediate.getLastTimestamp());
 predictionResults.addAll(intermediate.getPredictions());
}

return new DetectionPipelineResult(anomalies, DetectionUtils.consolidateNestedLastTimeStamps(lastTimeStamps), predictionResults);

```

Reply:
Got it! Appreciate your help with the code.